RIPS/rip-9999.md

```diff
@@ -0,0 +1,313 @@
+transaction's origin to be `keccak256(sender)[0..20]`.
+
+## Rationale
+
+### Authorization as a transaction subtype instead of a family of opcodes
+
+Both approaches have their benefits and downsides.
+Here are some of the benefits of a transaction subtype approach:
+
+* Use of `AUTH` and `AUTHCALL` opcodes only allow an unnecessarily dangerous way for EOAs to achieve Gas Abstraction.
+
+    While technically it is possible to implement a gas relaying feature on top of EIP-3074, this approach requires
+users to fully hand over control of their accounts to the gas paying "invoker" contract.
+A lot can go wrong with this approach.
+
+* Use of `AUTH` and `AUTHCALL` opcodes for Gas Abstraction introduces risks to the block building task.
+
+    If the block builder relies on a Gas Abstraction mechanism to pay for transaction gas, it will require a protection
+mechanism similar to [ERC-7562: Account Abstraction Validation Scope Rules](https://eips.ethereum.org/EIPS/eip-7562).
+
+    It is impossible to retrofit such mechanism into legacy transaction types.
+    There is no way to check the `AUTH` signature other than by executing it, which probably means that
+    there will always be a risk of the wrapper transaction reverting, leaving the relay service with a loss.
+
+* Avoids exposing ECDSA signature and account nonces to the EVM opcodes.
+    Using opcodes to verify authorization signature necessarily means that EVM becomes exposed to these details.
+    There are real issues with ECDSA signatures, so abstracting cryptography details from EVM opcodes is desired.
+
+* There is only a limited number of available opcodes across all L1 and L2 networks.
+
+    Transaction types are also limited, however with subtypes the namespace is basically unlimited.
+
+    This makes it easier to avoid network splits, as well as to introduce changes to authorisation behaviour if necessary.
+
+## Backwards Compatibility
+
+This RIP suggest creation of new RIP-7560 transaction subtypes, which by itself does not affect any existing contracts.
+
+However, any contract that assumes that an EOA may not have any code or storage, will have its expectations broken
+by any of these transaction subtypes. EOAs may be detected by performing an `ecrecover` or by accessing `tx.origin`. 
+
+## Security Considerations
+
+The `DELEGATED_EOA_SUBTYPE` and especially `AUTHORIZED_EOA_SUBTYPE` transactions add an ability for a contract to
+assume control over an EOA. This is a very powerful tool that has very serious security implications.
+
+Wallets must make sure the users know what they are signing and what the implications for the signing will be.
+Unlike other similar proposals that can rely on a fact that most wallet do not have an API to sign arbitrary data,
+these transaction subtypes could potentially be signed with a regular `eth_signTransaction` API of a wallet
+with RIP-7560 support.
```

Comment:
For that reason, I wonder if all these subtypes belong in the same RIP, or in a family of RIPs.  An L2 might choose to implement some of the subtypes while excluding others for security reasons.  WDYT?

Reply:
I agree but I wanted to avoid creating 4 new RIPs.
Should I do it?
The resulting RIPs will be somewhat similar and their 'motivation'/'abstract'/'rationale' sections will be pretty repetitive, only 'specification' will have any new information.

Reply:
Yes, they'll have a lot of similarity but it enables supporting some without supporting others.  I was hoping to make things optional whenever possible.  OTOH it's likely that they'll be adopted together anyway.